NEWS.md

```diff
@@ -1,80 +1,112 @@
+  (@clauswilke, #2991, #2525).
+
+* Re-generate economics data. This leads to some changes in the values of all
+  columns (especially in `psavert`), but more importantly, strips the grouping
+  attributes from `economics_long`.
 
 * Closed arrows in `element_line()` are now filled (@yutannihilation, #2924).
 
-* ggplot2 now works in Turkish locale (@yutannihilation, #3011).
+* Facet strips on the left side of plots now have clipping turned on, preventing
+  text from running out of the strip and borders from looking thicker than for
+  other strips (@karawoo, #2772 and #3061).
 
 * `GeomRibbon` and `GeomArea` now sort the data along the x-axis in the 
-  `setup_data()` method rather than as part of `draw_group()` (@thomasp85, #3023)
+  `setup_data()` method rather than as part of `draw_group()` (@thomasp85, 
+  #3023)
 
-* `coord_sf()`, `coord_map()`, and `coord_polar()` now squash `-Inf` and `Inf`
-  into the min and max of the plot (@yutannihilation, #2972).
+* ggplot2 now works in Turkish locale (@yutannihilation, #3011).
 
-* `stat_bin()` will now error when the number of bins exceeds 1e6 to avoid 
-  accidentally freezing the user session (@thomasp85).
+* ggplot2 no longer attaches any external packages when using functions that 
+  depend on packages that are suggested but not imported by ggplot2. The 
+  affected functions include `geom_hex()`, `stat_binhex()`, 
+  `stat_summary_hex()`, `geom_quantile()`, `stat_quantile()`, and `map_data()` 
+  (@clauswilke, #3126).
 
-* `stat_bin()` now handles data with only one unique value (@yutannihilation #3047).
+* `geom_hline()`, `geom_vline()`, and `geom_abline()` now throw a warning if the 
+  user supplies both an `xintercept`, `yintercept`, or `slope` value and a 
+  mapping (@RichardJActon, #2950).
 
-* `stat_function()` now accepts rlang/purrr style anonymous functions for the 
-  `fun` parameter (@dkahle, #3159).
+* `geom_rug()` now works with `coord_flip()` (@has2k1, #2987).
 
-* `geom_polygon()` can now draw polygons with holes using the new `subgroup` 
-  aesthetic. This functionality requires R 3.6 (@thomasp85, #3128)
+* Default labels are now generated more consistently; e.g., symbols no longer
+  get backticks, and long expressions are abbreviated with `...`
+  (@yutannihilation, #2981).
 
-* `x0` and `y0` is now recognized positional aesthetics so they will get scaled 
-  if used in extension geoms and stats (@thomasp85, #3168)
+* Layers now have a new member function `setup_layer()` which is called at the
+  very beginning of the plot building process and which has access to the 
+  original input data and the plot object being built. This function allows the 
+  creation of custom layers that autogenerate aesthetic mappings based on the 
+  input data or that filter the input data in some form. One example is the new 
+  `LayerSf` class which locates the geometry column in sf objects and sets up an 
+  aesthetic mapping for it (@clauswilke, #2872).
```

Comment:
Should we delete or modify this news item? As it currently stands, none of this code is user facing, and it cannot be used outside of ggplot2 proper. The main benefit at this time is a cleaner internal implementation of sf layers. @hadley @yutannihilation

Reply:
I think `setup_layer()` is the change that should be mentioned in NEWS, so I'm against deleting this. How about using the first sentence + "for internal use only"?

(Though I'm not sure if we can recommend this to ordinary users, but `setup_layer()` might be still useful by overwriting it from outside as you showed in this example: https://github.com/tidyverse/ggplot2/pull/2875#issuecomment-417835205)